Source/WebCore/page/PointerLockController.cpp

```diff
@@ -40,46 +42,84 @@
 
 namespace WebCore {
 
+class UnadjustedMovementPlatformMouseEvent : public PlatformMouseEvent {
+public:
+    UnadjustedMovementPlatformMouseEvent(const PlatformMouseEvent &src)
+        : PlatformMouseEvent(src.position(), src.globalPosition(), src.button(), src.type(), src.clickCount(), src.modifiers(), src.timestamp(), src.force(), src.syntheticClickType(), src.pointerId())
+    {
+        m_movementDelta = m_unadjustedMovementDelta = src.unadjustedMovementDelta();
+    }
+};
+
 PointerLockController::PointerLockController(Page& page)
     : m_page(page)
 {
 }
 
-void PointerLockController::requestPointerLock(Element* target)
+PointerLockController::~PointerLockController()
 {
+}
+
+void PointerLockController::requestPointerLock(Element* target, bool unadjustedMovement, RefPtr<DeferredPromise> promise)
+{
+#define REJECT(err) do { if (promise) { promise->reject(err); } } while (0)
+#define RESOLVE() do { if (promise) { promise->resolve(); } } while (0)
```

Comment:
I think it would be better to just check `if (promise)` where needed rather than use macros. Similar to `FullscreenManager::requestFullscreenForElement`.

Reply:
Done.